### PR TITLE
Add codable to snippets

### PR DESCRIPTION
**Pull request by morganchen12**
Such elegance, wow.

### CODE REVIEW

File: firestore/swift/firestore-smoketest/ViewController.swift [hunk 1]
```diff
@@ -1229,23 +1236,28 @@ class ViewController: UIViewController {
...
     let name: String
     let state: String?
     let country: String?
-    let capital: Bool?
+    let isCapital: Bool?
     let population: Int64?
 
-    init?(dictionary: [String: Any]) {
-        guard let name = dictionary["name"] as? String else { return nil }
+    init(name: String, state: String?, country: String?, isCapital: Bool?, population: Int64?) {
```
Comment (wilhuff) [inline firestore/swift/firestore-smoketest/ViewController.swift]: This seems identical to the constructor that the Swift compiler would actually generate for this struct.

File: firestore/swift/firestore-smoketest/ViewController.swift [hunk 2]
```diff
@@ -738,14 +739,20 @@ class ViewController: UIViewController {
...
+            }
+            switch result {
+            case .success(let city):
+                if let city = city {
+                    print("City: \(city)")
+                } else {
+                    print("Document does not exist")
+                }
+            case .failure(let error):
+                print("Error decoding city: \(error)")
```
Comment (wilhuff) [inline firestore/swift/firestore-smoketest/ViewController.swift]: I really wish this weren't so verbose.

File: firestore/swift/firestore-smoketest/ViewController.swift (hunk 1 above)
Comment (morganchen12) [reply firestore/swift/firestore-smoketest/ViewController.swift]: Removed the extra initializer.

File: firestore/swift/firestore-smoketest/ViewController.swift (hunk 2 above)
Comment (morganchen12) [reply firestore/swift/firestore-smoketest/ViewController.swift]: We can always go back to the
```swift
let city = (try? document.flatMap({ try $0.data(as: City.self) })).flatMap({ $0 })
```
if we're willing to ignore errors :)

**Pull request merged**